recipes/process-assert-to-assert/src/workflow.ts

```diff
@@ -0,0 +1,207 @@
+import { EOL } from 'node:os';
+import { getNodeRequireCalls } from '@nodejs/codemod-utils/ast-grep/require-call';
+import { getNodeImportStatements } from '@nodejs/codemod-utils/ast-grep/import-statement';
+import { resolveBindingPath } from '@nodejs/codemod-utils/ast-grep/resolve-binding-path';
+import { removeBinding } from '@nodejs/codemod-utils/ast-grep/remove-binding';
+import { removeLines } from '@nodejs/codemod-utils/ast-grep/remove-lines';
+import type {
+	Edit,
+	Range,
+	Rule,
+	SgNode,
+	SgRoot,
+} from '@codemod.com/jssg-types/main';
+import type JS from '@codemod.com/jssg-types/langs/javascript';
+
+type ReplaceRule = {
+	importNode?: SgNode<JS>;
+	binding?: string;
+	rule: Rule<JS>;
+	replaceWith?: string;
+};
+
+/**
+ * Transform function that converts deprecated `process.assert` usage to the `node:assert` module.
+ *
+ * Handles:
+ * 1. Replaces `process.assert(...)` member expressions/calls with `assert(...)` or `assert.xxx` as appropriate.
+ * 2. Handles cases where `process` is imported/required under a different binding (resolves binding paths).
+ * 3. Removes the original `process` import/require when it's only used for `assert` and removes the import line when empty.
+ * 4. Adds `import assert from "node:assert";` or `const assert = require("node:assert");` at the top
+ *    when the file does not already import/require `assert`.
+ *
+ * Steps:
+ * - Find all `process` import/require statements and resolve any binding for `assert`.
+ * - Replace call and member-expression usages that reference `process.assert` (or the resolved binding) with `assert`.
+ * - Remove or update the original import/require for `process` when it's no longer needed.
+ * - If `assert` is not already present, insert the appropriate `import` or `require` line depending on the module style.
+ *
+ * @param root - The AST root node provided by jssg for the file being transformed.
+ * @returns The transformed source code as a string, or `null` when no edits are required.
+ */
+export default function transform(root: SgRoot<JS>): string | null {
+	const rootNode = root.root();
+	const edits: Edit[] = [];
+	const linesToRemove: Range[] = [];
+	const replaceRules: ReplaceRule[] = [
+		{
+			rule: {
+				kind: 'member_expression',
+				pattern: 'process.assert',
+			},
+			replaceWith: 'assert',
+		},
+	];
+
+	const processImportsToRemove = new Set<SgNode<JS>>();
+
+	const requireCalls = getNodeRequireCalls(root, 'process');
+	const importStatements = getNodeImportStatements(root, 'process');
+	const allImports = [...requireCalls, ...importStatements];
+	const processUsages = rootNode.findAll({
+		rule: {
+			kind: 'member_expression',
+			has: {
+				kind: 'identifier',
+				regex: '^process$',
```

Comment:
nit: this can be simplified to
```suggestion
				pattern: 'process',
```